titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/v1/V1SpecPodFactory.java

```diff
@@ -291,6 +293,14 @@ private V1Container buildV1ExtraContainers(BasicContainer extraContainer) {
                 .volumeMounts(KubePodUtil.buildV1VolumeMounts(extraContainer.getVolumeMounts()));
     }
 
+    private Long getTerminationGracePeriodSeconds(Job<?> job) {
+        String jobKillWaitSeconds = job.getJobDescriptor().getAttributes().get(JOB_PARAMETER_ATTRIBUTES_KILL_WAIT_SECONDS);
+        if (jobKillWaitSeconds != null) {
+            return Long.valueOf(jobKillWaitSeconds);
```

Comment:
Try/catch in case a user sets invalid value. In the latter case, set default.

Reply:
added